docs/serving/spec/knative-api-specification-1.0.md

```diff
@@ -0,0 +1,2615 @@
+
+
+## Service
+
+When a Service is created, it MUST create and own a Configuration and Route with the same name as the Service. Updates to `spec`, `metadata.labels`, and `metadata.annotations` of the Service MUST be copied to the appropriate Configuration or Route, as follows:
+
+
+
+*   `metadata` changes MUST be copied to both Configuration and Route.
+*   In addition, the `serving.knative.dev/service` label on the Route and Configuration MUST be set to the name of the Service.
+*   Additional `labels` and `annotations` on the Configuration and Route not specified above MUST be removed.
+*   See the documentation of `spec` in the [detailed resource fields section](#detailed-resources--v1beta1) for the mapping of specific `spec` fields to the corresponding fields in Configuration and Route.
+
+Similarly, the Service MUST update its `status` fields based on the corresponding `status` of its owned Route and Configuration. The Service MUST include conditions of `ConfigurationsReady` and `RoutesReady` in addition to the generic `Ready` condition; other conditions MAY also be present.
+
+
+## Configuration
+
+When a Configuration is created or its `spec` updated, the following steps MUST be taken to create a new Revision (on creation, the previous state for all fields is an empty value):
+
+
+
+*   If the `spec.template` field has changed from the previous state, a new owned Revision MUST be created. If the Revision name is not provided by the user through the `spec.template.metadata.name` field, it MUST be system generated. The system generated name should be treated opaquely as no semantics are defined. The values of `spec.template.metadata` and `spec.template.spec` MUST be copied to the newly created Revision. If the Revision cannot be created, the Configuration MUST signal that the Configuration is not `Ready`.
+*   `metadata.labels` and `metadata.annotations` from the Configuration MUST NOT be copied to the newly-created Revision.
+
+Configuration MUST track the status of owned Revisions in order of creation, and report the name of the most recently created Revision in `status.latestCreatedRevisionName` and the name of the most recently created Revision where the `Ready` Condition is `True` in `status.latestReadyRevisionName`. These fields MAY be used by client software and Route objects to determine the status of the Configuration and the best target Revision for requests.
+
+
+## Revision
+
+A Revision is automatically scaled by the underlying infrastructure; a Revision with no referencing Routes MAY be automatically scaled to zero instances and backing resources collected. Additionally, Revisions which have been scaled to zero instances and which are older than the oldest live Revision (referenced by at least one Route or latest for the Configuration) MAY be automatically deleted by the system.
```

Comment:
nit: zero scale isn't a requirement for GC, only that things are orphaned.

Reply:
fixed